src/ratelimiter_map.rs

```diff
@@ -0,0 +1,135 @@
+use dashmap::{mapref::multiple::RefMulti, DashMap};
+use std::{env::var, sync::Arc};
+use tokio::time::{interval, Duration, Instant};
+use tracing::{debug, warn};
+use twilight_http::ratelimiting::Ratelimiter;
+
+pub struct RatelimiterMap {
+    default: Ratelimiter,
+    default_token: String,
+    max_size: Option<usize>,
+    inner: Arc<DashMap<String, (Ratelimiter, Instant)>>,
+}
+
+async fn reap_old_ratelimiters(map: Arc<DashMap<String, (Ratelimiter, Instant)>>) {
+    let client_reap_interval =
+        Duration::from_secs(var("CLIENT_REAP_INTERVAL").map_or(600, |timeout| {
+            if let Ok(timeout_secs) = timeout.parse() {
+                timeout_secs
+            } else {
+                warn!("Unable to parse CLIENT_REAP_INTERVAL, proceeding with defaults");
+                600
+            }
+        }));
+
+    let client_decay_timeout =
+        Duration::from_secs(var("CLIENT_DECAY_TIEOUT").map_or(3600, |timeout| {
+            if let Ok(timeout_secs) = timeout.parse() {
+                timeout_secs
+            } else {
+                warn!("Unable to parse CLIENT_DECAY_TIMEOUT, proceeding with defaults");
+                3600
+            }
+        }));
```

Comment:
This code is duplicated and could be split out to a function. Actually, we do environment variable parsing in multiple places in the application. It is probably best to have an Environment type that we instantiate on application initialization.